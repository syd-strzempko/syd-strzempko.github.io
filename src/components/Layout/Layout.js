import { Link } from 'react-router-dom';
import styles from './Layout.module.css';
import NavBar from '../NavBar';

const Layout = ({ header, children, returnURL }) => {
  return (
    <div className={styles.layout}>
      <NavBar />
      <div className={styles.contentWrapper}>
        <div className={styles.content}>
          {returnURL && <div className={styles.backWrapper}><Link to={returnURL} className={styles.backButton}>Â« back</Link></div>}
          <div className={styles.header}>{header}</div>
          <div className={styles.body}>{children}</div>
        </div>
      </div>
    </div>
  );
}

export default Layout;