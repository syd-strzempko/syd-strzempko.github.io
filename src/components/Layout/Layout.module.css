.layout {
    min-height: 100vh;
    width: 100%;
    display: flex;
    flex-flow: column nowrap;
    justify-content: center;
    z-index: 2;
    position: relative;
}
.backLink {
    position: fixed;
    top: 15px;
    left: 12px;
    z-index: 2;
    img {
        width: 42px;
        transform: rotate(180deg);
    }
}