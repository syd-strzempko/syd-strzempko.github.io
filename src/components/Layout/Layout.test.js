import Layout from '../Layout';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

// Note: MemoryRouter allows for nested Link components to exist inside of & access stack of mocked router

describe('Layout', () => {
  it('renders correctly', () => {
    let view = renderer.create(
      <MemoryRouter>
        <Layout />
      </MemoryRouter>
    );
    expect(view.toJSON()).toMatchSnapshot();
  });
});