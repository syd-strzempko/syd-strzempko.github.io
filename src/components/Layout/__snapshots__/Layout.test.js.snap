// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="layout"
>
  <div
    className="navWrapper"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/work"
      onClick={[Function]}
    >
      Work
    </a>
    <a
      href="/art"
      onClick={[Function]}
    >
      Art
    </a>
  </div>
  <div
    className="contentWrapper"
  >
    <div
      className="content"
    >
      <div
        className="header"
      />
      <div
        className="body"
      />
    </div>
  </div>
</div>
`;
